packages/core/content-manager/server/controllers/relations.js

```diff
@@ -186,39 +186,43 @@ module.exports = {
     }
 
     const queryParams = {
-      sort: mainField,
-      ...query,
-      fields: fieldsToSelect, // cannot select other fields as the user may not have the permissions
-      filters: {}, // cannot filter for RBAC reasons
+      select: fieldsToSelect,
     };
 
     if (!isEmpty(idsToOmit)) {
-      addFiltersClause(queryParams, { id: { $notIn: idsToOmit } });
-    }
-
-    // searching should be allowed only on mainField for permission reasons
-    if (_q) {
-      addFiltersClause(queryParams, { [mainField]: { $containsi: _q } });
+      queryParams.filters = { id: { $notIn: idsToOmit } };
     }
 
-    const subQuery = strapi.db.queryBuilder(sourceModel.uid);
-
-    const alias = subQuery.getAlias();
-
-    const knexSubQuery = subQuery
-      .where({ id, [`${alias}.id`]: { $notNull: true } })
-      .join({ alias, targetField })
-      .select(`${alias}.id`)
-      .getKnexQuery();
-
-    addFiltersClause(queryParams, { id: { $in: knexSubQuery } });
-
     if (MANY_RELATIONS.includes(attribute.relation)) {
-      ctx.body = await strapi.entityService.findPage(targetedModel.uid, queryParams);
+      const page = Number(query.page || 1);
```

Comment:
minor thing, why not setting a default value in the yup validations? It seems safer.

Reply:
It's now done in a lower level :) (database layer)